tiledb/api/src/datatype/physical.rs

```diff
@@ -330,3 +331,125 @@ where
         self.0.bits_cmp(&other.0)
     }
 }
+
+/// Represents a dynamically typed single physical value.
+///
+/// [PhysicalValue] holds the bits which correspond to a single value of a logical data type.
+/// For a given logical data type, [PhysicalValue] will always be constructed with the
+/// primitive numerical type of the same bit width, signed-ness, and precision.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum PhysicalValue {
```

Comment:
Does core provide any aggregates for string fields? I know we store some stuff in fragment metadata for string fields, but obvious sum aggregates wouldn't work there.

I don't want to derail this PR by going back to add string support if so, but I just wanted to note whether this is yet another case where we have an API where we basically need a "any physical type or a string" sort of enumeration. We seem to be collecting variations on this theme and at some point we'll probably want to go back and unify them all.

Reply:
From discussion:

yes, it does, one must set an offset buffer as they usually do.

This will go in a follow-up story.

Reply:
https://app.shortcut.com/tiledb-inc/story/54898/rs-add-support-for-running-min-max-aggregates-on-string-fields